src/Cache/Factory/DesarrollaCacheFactory.php

```diff
@@ -48,20 +48,38 @@ class DesarrollaCacheFactory implements FactoryInterface
     protected $config;
 
     /**
-     * @inheritdoc
+     * @param string|null $configFile
+     * @param array       $configArray
      */
-    public function make($config)
+    public function __construct($configFile = null, array $configArray = [])
     {
-        $this->config = $config['desarrolla'];
-        $driver = $this->getDriver();
+        // Default config from distribution
+        if (null === $configFile) {
+            $configFile = __DIR__.'/../../../config/config.php';
+        }
+
+        $config = include $configFile;
+
+        if (!isset($config['adapter']) || !isset($config['desarrolla']) || 'desarrolla' !== $config['adapter']) {
```

Comment:
I think, now since the config is used solely by adapter factory (and not the ratelimiter class that switcher adapters), it doesn't make sense to keep the $config['adapter']